core/include/scipp/core/variable.h

```diff
@@ -339,23 +346,39 @@ class SCIPP_CORE_EXPORT Variable {
   bool operator!=(const Variable &other) const;
   bool operator!=(const VariableConstProxy &other) const;
   Variable operator-() const;
+
   Variable &operator+=(const Variable &other) &;
   Variable &operator+=(const VariableConstProxy &other) &;
-  Variable &operator+=(const double value) &;
+  template <typename T, typename = std::enable_if_t<!is_variable_or_proxy<T>()>>
+  Variable &operator+=(const T v) & {
+    return *this += makeVariable<underlying_type_t<T>>(v);
+  }
+
   Variable &operator-=(const Variable &other) &;
   Variable &operator-=(const VariableConstProxy &other) &;
-  Variable &operator-=(const double value) &;
+  template <typename T, typename = std::enable_if_t<!is_variable_or_proxy<T>()>>
+  Variable &operator-=(const T v) & {
+    return *this -= makeVariable<underlying_type_t<T>>(v);
+  }
+
   Variable &operator*=(const Variable &other) &;
   Variable &operator*=(const VariableConstProxy &other) &;
-  Variable &operator*=(const double value) &;
+  template <typename T, typename = std::enable_if_t<!is_variable_or_proxy<T>()>>
+  Variable &operator*=(const T v) & {
+    return *this *= makeVariable<underlying_type_t<T>>(v);
+  }
   template <class T>
   Variable &operator*=(const boost::units::quantity<T> &quantity) & {
     setUnit(unit() * units::Unit(T{}));
     return *this *= quantity.value();
```

Comment:
Remove in favor of adding an overload of `makeVariable`?

Reply:
https://github.com/scipp/scipp/issues/585 will address this.